src/drivers/sifive_spi0.c

```diff
@@ -131,13 +137,36 @@ static int configure_spi(struct __metal_driver_sifive_spi0 *spi,
     return 0;
 }
 
+static void spi_mode_switch(struct __metal_driver_sifive_spi0 *spi,
+                            struct metal_spi_config *config,
+                            unsigned int qspi_mode) {
+    long control_base =
+        __metal_driver_sifive_spi0_control_base((struct metal_spi *)spi);
+
+    if (config->qspi_mode == qspi_mode) {
```

Comment:
Why is this check here? If they have to be equal, why not just use the one already in `config`?

Reply:
That' my mistake, I should not use the same name for the two variables.
I've renamed these two variables. 
First one is used for SPI configuration, the second is for tagging the data transfer stage.
The function spi_mode_switch will be active either after sending out cmd or before sending data.